internal/runners/shim/shim.go

```diff
@@ -0,0 +1,95 @@
+package shim
+
+import (
+	"os"
+	"path/filepath"
+
+	"github.com/ActiveState/cli/internal/fileutils"
+	"github.com/ActiveState/cli/internal/language"
+	"github.com/ActiveState/cli/internal/locale"
+	"github.com/ActiveState/cli/internal/logging"
+	"github.com/ActiveState/cli/internal/path"
+	"github.com/ActiveState/cli/internal/primer"
+	"github.com/ActiveState/cli/internal/scriptfile"
+	"github.com/ActiveState/cli/internal/subshell"
+	"github.com/ActiveState/cli/internal/virtualenvironment"
+	"github.com/ActiveState/cli/pkg/project"
+	"github.com/ActiveState/cli/pkg/projectfile"
+)
+
+type Shim struct {
+	subshell subshell.SubShell
+}
+
+type primeable interface {
+	primer.Outputer
+	primer.Subsheller
+}
+
+type Params struct {
+	Script   string
+	Language string
+}
+
+func New(prime primeable) *Shim {
+	return &Shim{
+		prime.Subshell(),
+	}
+}
+
+func (s *Shim) Run(params Params, args ...string) error {
+	var lang language.Language
+	if params.Language != "" {
+		lang = language.MakeByName(params.Language)
+	} else {
+		for _, l := range project.Get().Languages() {
+			// Use first language found
+			lang = language.MakeByName(l.Name())
+			break
+		}
+	}
+	if !lang.Recognized() {
+		return locale.NewError("err_shim_language", "Unsupported language")
+	}
```

Comment:
Language is irrelevant. We're just inheriting the environment. Whether the user wants to run `perl` on that or `whoami` does not matter to us.